contrib/inventory/ec2.py

```diff
@@ -469,7 +470,10 @@ def read_settings(self):
         cache_id = self.boto_profile or os.environ.get('AWS_ACCESS_KEY_ID', self.credentials.get('aws_access_key_id'))
         if cache_id:
             cache_name = '%s-%s' % (cache_name, cache_id)
-        cache_name += '-' + str(abs(hash(__file__)))[1:7]
+        if sys.version_info[0] == 2:
+            cache_name += '-' + hashlib.md5(__file__).hexdigest()[:6]
+        else:
+            cache_name += '-' + hashlib.md5(__file__.encode('utf-8')).hexdigest()[:6]
```

Comment:
Is there a reason you're not only using this line, but make this dependent on the Python version?

Reply:
Well, as I state in PR description, version in trunk practically disables ec2.py cache when using python 3. The idea was to replaces `hash()` call with any hashing function that gives stable results between script runs. Anything from hashlib module will do, `md5` does not have any advantage.

hashlib functions require byte string as input. Python 2 and 3 differ in what is stored in `__file__` variable: python 2 has simple string and in python 3 all strings are unicode.

Therefore in python 2 and 3 `__file__` variable has to be treated differently. Note that this is third commit in this PR, I could get past test from first attempt, and different lines were added in the process.

There is only one possible alternative to testing python version: testing for type of `__file__` variable. I chose testing for python version to simplify cleanup of old and unused code in future when python2 support will be dropped.

Reply:
> You also need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to).

OK, will do

Reply:
> Well, as I state in PR description, version in trunk practically disables ec2.py cache when using python 3. The idea was to replaces `hash()` call with any hashing function that gives stable results between script runs. Anything from hashlib module will do, `md5` does not have any advantage.

One important note: `md5` will not be available on some systems (such as FIPS compliant systems). It's probably better to use `sha256` (as `sha1` could also be removed).

> hashlib functions require byte string as input. Python 2 and 3 differ in what is stored in `__file__` variable: python 2 has simple string and in python 3 all strings are unicode.

You could do
```.py
from ansible.module_utils._text import to_bytes
```
and then
```.py
cache_name += '-' + hashlib.sha256(to_bytes(__file__)).hexdigest()[:6]
```
This will work for both Python 2 and Python 3.